### PR TITLE
Fix "redeclared without dllimport attribute..." warning for MinGW

### PR DESCRIPTION
**Pull request by ark0f**
#2356

### CODE REVIEW

File: CppParser/include/Poco/CppParser/CppParser.h [hunk 1]
```diff
@@ -31,7 +31,7 @@
 // CppParser_API functions as being imported from a DLL, whereas this DLL sees symbols
 // defined with this macro as being exported.
 //
-#if (defined(_WIN32) || defined(__CYGWIN__)) && defined(POCO_DLL)
+#if (defined(_MSC_VER) || defined(__CYGWIN__)) && defined(POCO_DLL)
```
Comment (aleks-f) [inline CppParser/include/Poco/CppParser/CppParser.h]: @ark0f these should probably be properly checked for MSVC compiler, see #2428 why and how it was fixed in https://github.com/pocoproject/poco/commit/5de2bf7ecffa35b1ecdc07a147942b0c6ff9162a

Comment (ark0f) [reply CppParser/include/Poco/CppParser/CppParser.h]: I understood. It will be fixed

**Comment by aleks-f** [comment]
@ark0f this tidy-up is a good effort, but it seems something caused many travis failures, those need to be taken care of before merging this

**Comment by ark0f** [comment]
Thank you. I know about Travis. I see that jobs were interrupted and I can't view full logs as it are more than 4 MB (Travis log limit). I can try to use `git commit --amend` to restart build totally.

**Comment by ark0f** [comment]
Restart didn't give result.

**Comment by ark0f** [comment]
Hm... I see that Clang produce many warnings `unused typedef`. I'll try to fix it.

**Comment by ark0f** [comment]
@aleks-f, I think we need to refactor `POCO_COMPILER_CLANG` definition to avoid problems with using `POCO_COMPILER_GCC` instead of `__GNUC__` as Clang is GCC-like compiler (like MinGW).

**Pull request merged**